npm/private/npm_import.bzl

```diff
@@ -501,11 +501,19 @@ def _download_and_extract_archive(rctx):
             fail(msg)
 
 def _npm_import_rule_impl(rctx):
+    has_lifecycle_hooks = not (not rctx.attr.lifecycle_hooks) or not (not rctx.attr.custom_postinstall)
+    has_patches = not (not rctx.attr.patches)
+
+    package_src = _EXTRACT_TO_DIRNAME
     if utils.is_git_repository_url(rctx.attr.url):
         _fetch_git_repository(rctx)
     else:
         _download_and_extract_archive(rctx)
```

Comment:
TODO: if using `package_src = _TARBALL_FILENAME` then we only need to extract the package.json file and not the whole archive

Reply:
Trying this now. Ready for review and landing otherwise.

Reply:
Done and working